build-logic/src/main/kotlin/minestom.common-conventions.gradle.kts

```diff
@@ -0,0 +1,28 @@
+plugins {
+    `java-library`
+}
+
+configurations.all {
+    // We only use Jetbrains Annotations
+    exclude("org.checkerframework", "checker-qual")
+}
+
+java {
+    toolchain {
+        languageVersion.set(JavaLanguageVersion.of(17))
+    }
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
+    withJavadocJar()
+    withSourcesJar()
+}
+
+tasks {
+    withType<JavaCompile> {
+        sourceCompatibility = JavaVersion.VERSION_17.toString()
+        targetCompatibility = JavaVersion.VERSION_17.toString()
```

Comment:
Don't you already set these above in the `java` configuration? Pretty sure you don't need to repeat yourself here.